### PR TITLE
Add support for forbidding features when creating or modifying some domains

### PR DESCRIPTION
**Pull request by iliajie**
Hey Jamie!

As discussed in [this thread](https://forum.virtualmin.com/t/new-installer-questions-about-the-default-ssl-certificate/126684/10?u=ilia) this PR protects users from enabling mail for a virtual server that matches hostname.

### CODE REVIEW

**Comment by jcameron** [comment]
Is having email enabled when the domain name matches the hostname really not allowed? The linked thread doesn't really go into this at all ..

**Comment by iliajie** [comment]
> Is having email enabled when the domain name matches the hostname really not allowed? The linked thread doesn't really go into this at all ..

This is something [Joe](@swelljoe) has always advised against! It didn't just come up in this thread this time.

**Comment by verne-work** [comment]
will there be a way to override it somehow, someplace?   Not for mail, but we create a virtual server to have apache respond to the server's name so our external Nagios monitoring system can test if apache is running or not.

**Comment by iliajie** [comment]
It won't display mail-related options only, and only if they aren't enabled. 

It will be possible to use Virtualmin CLI to force enable those features.

**Comment by jcameron** [comment]
@verne-work it will still be possible to have a domain with a website enabled that matches the hostname

File: virtual-server-lib-funcs.pl [hunk 1]
```diff
@@ -19708,6 +19708,35 @@ sub list_possible_domain_features
...
+return @rv;
+}
+
+# filter_possible_domain_features(&features, &domain, [creating])
+# Given a list of features, returns only those that aren't forbidden for domain
+sub filter_possible_domain_features
+{
+my ($features, $d, $new) = @_;
+my @forbidden = &forbidden_domain_features($d, $new);
+@$features = grep {
```
Comment (jcameron) [inline virtual-server-lib-funcs.pl:19735]: Why not return the new list instead of updating the array reference?

File: domain_setup.cgi [hunk 2]
```diff
@@ -457,6 +457,11 @@ $cerr = &virtual_server_clashes(\%dom);
 $lerr = &virtual_server_limits(\%dom);
 &error($lerr) if ($lerr);
 
+# Check if features are not forbidden
+foreach my $ff (&forbidden_domain_features(\%dom, 1)) {
+	$dom{$ff} && &error(&text('setup_efeatforbidhostdef', $ff));
```
Comment (jcameron) [inline domain_setup.cgi:462]: This message is very specific to the error of the domain name matching the hostname, but couldn't forbidden features apply in other situations?

File: virtual-server-lib-funcs.pl (hunk 1 above)
Comment (iliajie) [reply virtual-server-lib-funcs.pl:19735]: It's a better practice in Perl. We do it all the time.

File: domain_setup.cgi (hunk 2 above)
Comment (iliajie) [reply domain_setup.cgi:462]: What other situations? Oh, you mean those that don't potentially exist yet?

Comment (iliajie) [reply domain_setup.cgi:462]: Currently it's only meant to be used with default host domain.. if there is something else, we'll adjust an error message.

**Pull request merged**